modin/core/dataframe/pandas/exchange/dataframe_protocol/from_dataframe.py

```diff
@@ -248,38 +278,49 @@ def datetime_column_to_ndarray(col: ProtocolColumn) -> Tuple[np.ndarray, Any]:
     dbuf, dtype = buffers["data"]
     # Consider dtype being `uint` to get number of units passed since the 01.01.1970
     data = buffer_to_ndarray(
-        dbuf, (DTypeKind.UINT, dtype[1], "u", "="), col.offset, col.size
+        dbuf,
+        (
+            DTypeKind.UINT,
+            dtype[1],
+            getattr(ArrowCTypes, f"UINT{dtype[1]}"),
+            Edianness.NATIVE,
+        ),
+        col.offset,
+        col.size,
     )
 
-    if format_str.startswith("ts"):
+    def parse_format_str(format_str, data):
+        """Parse datetime `format_str` to interpret the `data`."""
         # timestamp 'ts{unit}:tz'
-        meta = format_str[2:].split(":")
-        if len(meta) == 1:
-            unit = meta[0]
-            tz = ""
-        else:
-            unit, tz = meta
-        if tz != "":
-            raise NotImplementedError("Timezones are not supported yet")
-        if unit != "s":
-            # the format string describes only a first letter of the unit, add one extra
-            # letter to make the unit in numpy-style: 'm' -> 'ms', 'u' -> 'us', 'n' -> 'ns'
-            unit += "s"
-        data = data.astype(f"datetime64[{unit}]")
-    elif format_str.startswith("td"):
+        timestamp_meta = re.findall(r"ts([smun]):(.*)", format_str)
```

Comment:
Is `:` required to be present? I mean, isn't `tsm` a valid timestamp?.. also why `.findall()` instead of `.search()`?
Do note that this pattern would work for timestamps like `aaa123 tsm:qqq asdjklhffdkajh`, which should not be so.

I think the correct regexp should be something like
```python
m = re.match(r"ts([smun])(:(.*)|$)", format_str)
if m:
    unit, tz = m.group(1), m.group(3)
```

Reply:
> Is : required to be present?

Yes. Arrow documentation says ":" must always be present [[1] Notes 1](https://arrow.apache.org/docs/format/CDataInterface.html#data-type-description-format-strings):

> The timezone string is appended as-is after the colon character :, without any quotes. If the timezone is empty, the colon : must still be included.

So replaced `.findall()` with `.match()` without changing the regexp
```diff
-        timestamp_meta = re.findall(r"ts([smun]):(.*)", format_str)
+        timestamp_meta = re.match(r"ts([smun]):(.*)", format_str)
        if timestamp_meta:
-            unit, tz = timestamp_meta[0]
+            unit, tz = timestamp_meta.group(1), timestamp_meta.group(2)
```